pfDataSource.Test/DataSourceConfigurationServiceTests.cs

```diff
@@ -0,0 +1,189 @@
+
+using FluentAssertions;
+using System.Text;
+using pfDataSource.Common;
+using pfDataSource.Common.Configuration;
+using pfDataSource.Services;
+using Moq;
+using pfDataSource.Db;
+using pfDataSource.Db.Models;
+using System.Security.Cryptography;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace pfDataSource.Test
+{
+    public class DataSourceConfigurationServiceTests
+    {
+        private SourceConfiguration _sourceConfiguration;    
+
+        public DataSourceConfigurationServiceTests() {
+
+            _sourceConfiguration = new SourceConfiguration
+            {
+                Id = 1,
+                SourceType = "test-type",
+                DisplayType = "test",
+                DisplayName = "Test",
+                TempFilesPath = "/local/dir",
+                PureFarmingSourceName = "Test Source",
+                PureFarmingFullSourceName = "Full Test Source",
+                AwsSecrectId = "123456789",
+                AwsSecretKey = "some-secret-key",
+                AwsS3BucketArn = "aws::arn"
+            };
+
+        }
+
+
+        [Fact]
+        public void DataSourceConfigurationServiceTests_BuildConfigurationObject_Returns_Valid_SourceConfiguration()
+        {
+            // Test without configuration object
+           
+            var dataSourceConfigurationServiceMock = new DataSourceConfigurationService();
+
+            var result = dataSourceConfigurationServiceMock.BuildConfigurationObject(_sourceConfiguration, _sourceConfiguration.AwsSecrectId, _sourceConfiguration.AwsSecretKey);
+
+            result.Should().NotBeNull();
+
+            result.SourceType.Should().Be("test-type");
+            result.DisplayType.Should().Be("test");
+            result.DisplayName.Should().Be("Test");
+            result.TempFilesPath.Should().Be("/local/dir");
+            result.Name.Should().Be("Test Source");
+            result.FullName.Should().Be("Full Test Source");
+            result.Aws.S3BucketArn.Should().Be("aws::arn");
+            result.Aws.SecretId.Should().Be("123456789");
+            result.Aws.SecretKey.Should().Be("some-secret-key");
+            result.Configuration.Should().BeNull();
+
+            // Test with configuration object
+            _sourceConfiguration.Configuration = JsonConvert.SerializeObject(new { prop1 = "value1", prop2 = "value2" });
+
+            var result2 = dataSourceConfigurationServiceMock.BuildConfigurationObject(_sourceConfiguration, _sourceConfiguration.AwsSecrectId, _sourceConfiguration.AwsSecretKey, typeof(object));
+
+            result2.Should().NotBeNull();
```

Comment:
Should these second sets of asserts be seperate test cases?

Reply:
Sure, I can pull those out into another case.